src/transcoding.ts

```diff
@@ -309,7 +346,16 @@ export function transcode(
     }
   }
   // request is supposed to have keys in camelCase.
-  const snakeRequest = requestChangeCaseAndCleanup(request, camelToSnakeCase);
+  let fieldsToChange = undefined;
+  if (requestFields) {
+    fieldsToChange = getAllFieldNames(requestFields, []);
+  }
+  const snakeRequest = requestChangeCaseAndCleanup(
+    request,
+    camelToSnakeCase,
+    fieldsToChange
```

Comment:
The set should be built here, otherwise, you keep making the set from the list on each invocation of the recursive function. You can just do it once here and pass it to the `requestChangeCaseAndCleanup`.

Reply:
@alexander-fenster, Set does not have the `map` function we use in line 246. So our options are to either take a Set as a parameter and iterate through the elements with a loop and reassign, or else what we have here - what would you prefer?